### PR TITLE
Refactor FXIOS-7640 [v121] OrientationLockUtility in DefaultBrowserOnboardingViewController

### PR DESCRIPTION
**Pull request by FilippoZazzeroni**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7640)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/17042)

## :bulb: Description
Remove `OrientationLockUtility` in `DefaultBrowserOnboardingViewController`.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [x] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by yoanarios** [comment]
thanks for the contribution @FilippoZazzeroni, the build failed because of a new warning at `DefaultBrowserOnboardingViewController` here are the details:
```
Lines should not have trailing whitespace
Warning:Line:277
```

**Comment by lmarceau** [comment]
@FilippoZazzeroni I think with this change `OrientationLockUtility` isn't used anymore? If so it means we can remove it 😄

**Comment by FilippoZazzeroni** [comment]
@yoanarios you're welcome, i'll remove it. @lmarceau yes sure, i'm going to check it.

**Comment by FilippoZazzeroni** [comment]
@yoanarios @lmarceau everything should be done.

File: Client/Coordinators/Launch/LaunchCoordinator.swift [hunk 1]
```diff
@@ -115,7 +115,11 @@ class LaunchCoordinator: BaseCoordinator,
         defaultOnboardingViewController.preferredContentSize = CGSize(
             width: ViewControllerConsts.PreferredSize.DBOnboardingViewController.width,
             height: ViewControllerConsts.PreferredSize.DBOnboardingViewController.height)
-        defaultOnboardingViewController.modalPresentationStyle = .formSheet
+        if UIDevice.current.userInterfaceIdiom == .phone {
```
Comment (jnrahme) [inline Client/Coordinators/Launch/LaunchCoordinator.swift:118]: might be better to use Swift's conditional operator 
'defaultOnboardingViewController.modalPresentationStyle = (UIDevice.current.userInterfaceIdiom == .phone) ? .fullScreen : .formSheet'

File: Client/Frontend/DefaultBrowserOnboarding/DefaultBrowserOnboardingViewController.swift [hunk 2]
```diff
@@ -126,19 +126,9 @@ class DefaultBrowserOnboardingViewController: UIViewController, OnViewDismissabl
...
-        // Portrait orientation: lock enable
-        OrientationLockUtility.lockOrientation(UIInterfaceOrientationMask.portrait,
-                                               andRotateTo: UIInterfaceOrientation.portrait)
-    }
-
-    override func viewWillDisappear(_ animated: Bool) {
-        super.viewWillDisappear(animated)
-        // Portrait orientation: lock disable
-        OrientationLockUtility.lockOrientation(UIInterfaceOrientationMask.all)
+        if #available(iOS 16, *) {
```
Comment (jnrahme) [inline Client/Frontend/DefaultBrowserOnboarding/DefaultBrowserOnboardingViewController.swift:129]: same here  you can use a conditional statement in a single line using the nil coalescing operator
single-line expression are more concise and may be preferred for simple availability checks where you want to keep the code short
#available(iOS 16, *) ? setNeedsUpdateOfSupportedInterfaceOrientations() : ()

**Comment by FilippoZazzeroni** [comment]
@jnrahme i corrected everything

**Pull request merged**